app/assets/javascripts/leaflet.map.js

```diff
@@ -32,6 +32,11 @@ L.OSM.Map = L.Map.extend({
           layerOptions.apikey = OSM[value];
         } else if (property === "leafletOsmId") {
           layerConstructor = L.OSM[value];
+        } else if (property === "leafletOsmDarkId" && OSM.isDarkMap() && L.OSM[value]) {
+          layerConstructor = L.OSM[value];
+          layerOptions.className += " dark";
+        } else if (property === "filterClass" && OSM.isDarkMap()) {
```

Comment:
I think something like `darkFilterClass` would be better as the property name, so that it's clear what the filtering is for.